crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -214,4 +208,81 @@ impl QueryLimitsChecker {
 
         Ok(())
     }
+
+    fn analyze_selection_set(
+        &self,
+        cfg: &ServiceConfig,
+        fragment_defs: &HashMap<Name, Positioned<FragmentDefinition>>,
+        sel_set: &Positioned<SelectionSet>,
+    ) -> ServerResult<ComponentCost> {
+        // Use BFS to analyze the query and
+        // count the number of nodes and the depth of the query
+
+        // Queue to store the nodes at each level
+        let mut que = VecDeque::new();
+        // Number of nodes in the query
+        let mut num_nodes: u32 = 0;
+        // Depth of the query
+        let mut depth: u32 = 0;
+        // Number of nodes at each level
+        let mut level_len;
```

Comment:
nit: can we not declare this just below, at the time we initialise it?  Leaving it undefined seems to imply it could be initialised along multiple paths.